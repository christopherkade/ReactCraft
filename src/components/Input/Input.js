import React from "react"
import styled from "styled-components"

const Label = styled.span`
  font-family: Arial, Helvetica, sans-serif;
  position: absolute;
  color: white;
  margin-left: 0.5rem;
  margin-top: 5px;
  font-weight: lighter;
`

const InputBox = styled.input`
  font-family: Arial, Helvetica, sans-serif;
  box-shadow: inset 1px 1px 7rem 0rem black;
  border: 3px ridge darkgrey;
  background-color: rgb(47, 27, 37, 0.8);
  color: white;
  border-radius: 2px;
  padding: 2px 2px 2px 2.5rem;
  font-size: 1rem;

  &:focus {
    outline: transparent;
  }
`

const Input = () => {
  return (
    <div>
      <Label>Say: </Label>
      <InputBox />
    </div>
  )
}

export default Input
