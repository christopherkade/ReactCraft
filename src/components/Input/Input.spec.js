import React from 'react';
import Input from './Input';
import { render } from 'react-testing-library';

describe('<Input />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Input />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('input label', () => {
    it('should display Say: as the input\'s label', () => {
      const { container } = render(<Input />);
      expect(container.querySelector("label").textContent).toBe('Say: ')
    })
  })
});
