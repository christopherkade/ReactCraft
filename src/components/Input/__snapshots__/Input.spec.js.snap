// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should match snapshot 1`] = `
<div>
  <div>
    <span
      class="sc-bdVaJa klgfqw"
    >
      Say: 
    </span>
    <input
      class="sc-bwzfXH iBLEiv"
    />
  </div>
</div>
`;
